src/ansys/fluent/visualization/graphics/graphics_windows_manager.py

```diff
@@ -933,8 +932,10 @@ def _open_and_plot_console(
             self._grid = grid
 
         if not self._plotter_thread:
-            if FluentConnection._monitor_thread:
-                FluentConnection._monitor_thread.cbs.append(self._exit)
+            if obj is not None:
+                obj.get_root().session._fluent_connection._monitor_thread.cbs.append(
```

Comment:
@prmukherj `_monitor_thread` is actually a class attribute of `FluentConnection` which would make the above code unnecessary. It would be replaced by `FluentConnection._monitor_thread.cbs.append(...)`. However, I don't understand _why_ it's a class attribute--don't we need one per Fluent session? We should also bring it to the top (i.e. _Session_ level) where it would be public and give it an `add_on_exit()` method.

Reply:
@seanpearsonuk, actually I wanted to reduce the direct imports as much as possible and thus this change. But yes we can get back to the original. Regarding the other suggestion, yes we can do something like that.

Reply:
This is resolved at par suggestion.